src/Features/CSharp/Portable/ConvertProgram/ConvertProgramCodeFixProvider.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertProgram
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.ConvertToProgramMain), Shared]
+    internal class ConvertToProgramMainCodeFixProvider : SyntaxEditorBasedCodeFixProvider
```

Comment:
That should be in `Analyzers` layer.

Reply:
this matches the ConvertNamespace fixer work.  we can move both over at the same time if appropriate.

Reply:
@CyrusNajmabadi This causes bugs like https://github.com/dotnet/roslyn/issues/45569

There are many codefixes that are misplaced in `Features`. I think it's better if future codefixes are placed properly making the future work to move everything to `Analyzers` more manageable.

Reply:
so, i'm amenable to that, but i think ti should be done as a general work item that also ensures that the analyzer/fixer layer has all the appropraite other extensions/modules needed to make that work.